packages/ui/src/MembersView/index.tsx

```diff
@@ -246,19 +246,55 @@ export default function MembersView({
     };
   }, []);
 
+  const validateEmail = (emailString: string): boolean => {
+    // Regular expression pattern to validate email addresses
+    const emailPattern = /^[\w-]+(\.[\w-]+)*@([\w-]+\.)+[a-zA-Z]{2,7}$/;
+
+    // Split the email string by commas
+    const emails = emailString.split(',');
+
+    for (let i = 0; i < emails.length; i++) {
+      const email = emails[i].trim(); // Remove leading/trailing whitespace
+
+      if (!emailPattern.test(email)) {
+        return false; // Invalid email address
+      }
+    }
+
+    return true; // All email addresses are valid
+  };
+
   async function createInvite(event: any) {
     event.preventDefault();
     setLoading(true);
     try {
       const form = event.target;
-      const email = form.email.value;
+      const emailList = form.email.value;
       const role = form.role.value;
-      await api.createInvite({
-        email,
-        role,
-        communityId: currentCommunity.id,
-      });
-      routerReload();
+
+      const alreadyInvitedUserEmails = users.map((user) => user.email);
+      const emails = emailList.split(',');
+
+      // Discard already invited email address
+      const uniqueEmails = emails.filter(
```

Comment:
Can we reorder the code a little bit, so that error handling is done first?

```js
const emails = form.email.value.split(',')
const existingEmails = users.map((user) => user.email)
const usedEmails = emails.filter((email) => existingEmails.includes(email)) // or add an intersection method in `utilities/array`

if (usedEmails.length > 0) {
   return Toast.error(`Provided emails are already used: ${usedEmails.join(', ')}`)
}
if (!isEmailValid(emails.join(',')) {
  return Toast.error('Please enter a valid email')
}
await api.createInvite({
  email: uniqueEmails.join(','),
  role,
  communityId: currentCommunity.id,
});
routerReload();
```